trieste/utils/pareto.py

```diff
@@ -105,3 +105,39 @@ def _bounds_2d(front: TensorType) -> BoundedVolumes:
         upper = tf.concat([range_ + 1, pf_ext_idx[::-1, 1:][: pf_ext_idx[-1, 0]]], axis=-1)
 
         return BoundedVolumes(lower, upper)
+
+    def hypervolume_indicator(self, reference: TensorType) -> TensorType:
+        """
+        Calculate the hypervolume indicator
+        The hypervolume indicator is the volume of the dominated region.
+
+        :param reference: a reference point to use, with shape [D].
+            Should be equal or bigger than the anti-ideal point of the Pareto set.
```

Comment:
victor mentioned it's useful to be able to specify a reference point with elements less than those of the anti-ideal point in some dimensions. That this can be used when treating objectives as constraints, or when you want to focus on a particular section of the front. This algorithm appears to work in that case as long as there is at least one pareto point beyond the reference point, but it fails when the reference point is beyond the pareto front. In that scenario, victor and I think it makes sense to have a volume of zero

Reply:
actually, let's just stick with it as it is, and error if it's not greater than or equal to the anti-ideal point. I think that's just
```
tf.debugging.assert_greater_equal(reference, self.front)
```

Reply:
we can add the other functionality later

Reply:
okay, maybe better to open an issue for it.

Reply:
yeah. I can do that

Reply:
thanks